portfolio/src/main/webapp/script.js

```diff
@@ -127,14 +127,21 @@ $("#oldest").click(() => sortPhotos("oldest"));
  * fetch text from /data to display
  */
 function getFetchRequest() {
-  fetch("/data").then((response) =>
-    response.text().then((txt) => $("#fetch-response").append(txt))
-  );
+  fetch("/data").then(response => response.json()).then((json) => {
+    for (const comment of json) {
+      const component = $(`
+        <div>
+          ${comment.location}
+          <a href=${comment.link} target=\"_blank\">Learn more</a>
```

Comment:
just for security, any `target="_blank"` link should have `rel="noopener noreferrer"`
https://web.dev/external-anchors-use-rel-noopener/